### PR TITLE
Indexer Sample Code

### PR DESCRIPTION
**Pull request by ryanRfox**
This PR includes content updates for
- features/indexer

Changes include
- Adding code samples for all SDKs + cURL

### CODE REVIEW

**Comment by ryanRfox** [comment]
The code samples are "done" and ready for review. I intend to add references to the code samples (pending #12) near the top and bottom of the page.

File: docs/features/indexer.md [hunk 1]
```diff
@@ -7,16 +7,85 @@ This guide explains how to search the Algorand Blockchain using the V2 Indexer.
...
+
+	// instantiate indexer client
+   indexerClient, err := indexer.MakeClient(indexerAddress, indexerToken)
+}
+```
 
 !!! info 
     When using cURL be aware that the parameters may need to be URL encoded. The SDKs handle the encoding of parameter data. 
 
+Before describing each of the available REST APIs, a few specific functions are described that are supported across many of the calls. This includes [Paginated Results](#paginated-results), [Historical Data searches](#historical-data-searches), and [Note Field Seaching](#note-field-searching).
```
Comment (ejbaran) [inline docs/features/indexer.md:87]: typo on "Note Field Seaching" -> should be "Note Field Searching"

File: docs/features/indexer.md [hunk 2]
```diff
@@ -27,24 +96,127 @@ When searching large amounts of blockchain data often the results may be too lar
...
+result, err := indexerClient.SearchForTransactions().CurrencyGreaterThan(minAmount).Do
+
+// Print results
+result, err := json.MarshalIndent(transactions, "", "\t")
+fmt.Printf(string(JSON) + "\n")
+```
 
 ```bash tab="cURL"
- curl localhost:8980/v2/transactions?currency-greater-than=10
+ curl "localhost:8980/v2/transactions?currency-greater-than=10"
```
Comment (ejbaran) [inline docs/features/indexer.md:151]: Since this is CLI, can we add "$ " to the beginning of the line. Do this for all examples.

**Comment by ryanRfox** [comment]
@ejbaran the latest commits should resolve your comments above. Note the info boxes at the top an bottom now include links to the repo and download file (to-be included in #12).

**Comment by ejbaran** [comment]
Nice @ryanRfox . The only thing is the link to the example folder might have to be https://github.com/algorand/docs/tree/master/examples/indexer . That was my bad I think I gave you that other link :)

**Comment by ryanRfox** [comment]
Good point @ejbaran I now updated the links as if there are in production. To test, please replace /algorand/.../master with /ejbaran/.../raf-sample-code-directories

**Pull request merged**